MdeModulePkg/Bus/Pci/NvmExpressDxe/UnitTest/MediaSanitizeUnitTest.c

```diff
@@ -0,0 +1,1208 @@
+  //  IN UINT32                   MediaId,
+  //  IN UINT32                   PurgeAction,
+  //  IN UINT32                   OverwritePattern
+  //  );
+
+ #if 1
+  Status = NvmExpressMediaPurge (
+             &NvmeDevice->MediaSanitize,
+             NvmeDevice->Media.MediaId,
+             PurgeAction,
+             OverwritePattern
+             );
+
+  UT_ASSERT_NOT_EFI_ERROR (Status);
+ #endif
+
+ #if 0
+  //
+  // Case 2: Crypto Erase
+  //
+  SanitizeAction            = SANITIZE_ACTION_CRYPTO_ERASE;
+  NoDeallocateAfterSanitize = 0;
+  OverwritePattern          = 0;
+
+  Status = NvmExpressSanitize (
+             &NvmeDevice->BlockIo,
+             NamespaceId,
+             SanitizeAction,
+             NoDeallocateAfterSanitize,
+             OverwritePattern
+             );
+
+  UT_ASSERT_NOT_EFI_ERROR (Status);
+
+  //
+  // Case 3: Overwrite
+  //
+  SanitizeAction            = SANITIZE_ACTION_OVERWRITE;
+  NoDeallocateAfterSanitize = 0;
+  OverwritePattern          = 0xDEADBEEF;
+
+  Status = NvmExpressSanitize (
+             &NvmeDevice->BlockIo,
+             NamespaceId,
+             SanitizeAction,
+             NoDeallocateAfterSanitize,
+             OverwritePattern
+             );
+
+  UT_ASSERT_NOT_EFI_ERROR (Status);
+
+  //
+  // Case 4: Block Erase (invalid overwrite pattern)
+  //
+  SanitizeAction            = SANITIZE_ACTION_BLOCK_ERASE;
+  NoDeallocateAfterSanitize = 0;
+  OverwritePattern          = 0xDEADBEEF;
+
+  Status = NvmExpressSanitize (
+             &NvmeDevice->BlockIo,
+             NamespaceId,
+             SanitizeAction,
+             NoDeallocateAfterSanitize,
+             OverwritePattern
+             );
+
+  UT_ASSERT_STATUS_EQUAL (Status, EFI_INVALID_PARAMETER);
+
+  //
+  // Case 5: Overwrite (invalid overwrite pattern)
+  //
+  SanitizeAction            = SANITIZE_ACTION_OVERWRITE;
+  NoDeallocateAfterSanitize = 0;
+  OverwritePattern          = 0;
+
+  Status = NvmExpressSanitize (
+             &NvmeDevice->BlockIo,
+             NamespaceId,
+             SanitizeAction,
+             NoDeallocateAfterSanitize,
+             OverwritePattern
+             );
+
+  UT_ASSERT_STATUS_EQUAL (Status, EFI_INVALID_PARAMETER);
+ #endif
+
+  UnitTestStatus = NvmeDestroyDeviceInstance (&NvmeDevice);
+
+  DEBUG ((DEBUG_VERBOSE, "[          ] MediaSanitizePurgeUnitTest: Passed!!!!\n"));
+
+  return UNIT_TEST_PASSED;
+}
+
+UNIT_TEST_STATUS
+EFIAPI
+NvmeSanitizeUnitTest (
+  IN UNIT_TEST_CONTEXT  Context
+  )
+{
+  UINT32                    NamespaceId = 0;
+  UINT32                    SanitizeAction;
+  UINT32                    NoDeallocateAfterSanitize;
+  UINT32                    OverwritePattern;
+  UNIT_TEST_STATUS          UnitTestStatus = UNIT_TEST_PASSED;
+  NVME_DEVICE_PRIVATE_DATA  *NvmeDevice    = NULL;
+  EFI_STATUS                Status         = EFI_SUCCESS;
+
+  DEBUG ((DEBUG_VERBOSE, "[          ] NvmeSanitizenitTest: Starting Unit Test\n"));

```

Comment:
```suggestion
  DEBUG ((DEBUG_VERBOSE, "[          ] NvmeSanitizeUnitTest: Starting Unit Test\n"));

```

Also, it might be good to switch to use below to save the duplicated function name spell-outs.
```suggestion
  DEBUG ((DEBUG_VERBOSE, "%a: Starting Unit Test\n", __FUNCTION__));

```

Reply:
Agree. Will update to use __FUNCTION__.